plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/clients/admin/SslKafkaAdminFactory.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.kafka.clients.admin;
+
+import io.trino.plugin.kafka.security.KafkaSecurityConfig;
+import io.trino.spi.connector.ConnectorSession;
+
+import javax.inject.Inject;
+
+import java.util.Properties;
+
+import static java.util.Objects.requireNonNull;
+
+public class SslKafkaAdminFactory
+        implements KafkaAdminFactory
+{
+    private final Properties kafkaProperties;
+
+    @Inject
+    public SslKafkaAdminFactory(PlainTextKafkaAdminFactory plaintextAdminFactory, KafkaSecurityConfig securityConfig)
+    {
+        requireNonNull(plaintextAdminFactory, "plaintextAdminFactory is null");
+        requireNonNull(securityConfig, "securityConfig is null");
+
+        kafkaProperties = plaintextAdminFactory.configure(null);
```

Comment:
do not pass null here. Make a call withing `configure` and pass session.

Sth like this should do:
```java
public class SslKafkaAdminFactory
        implements KafkaAdminFactory
{
    private final Properties securityProperties;
    private final PlainTextKafkaAdminFactory plaintextAdminFactory;

    @Inject
    public SslKafkaAdminFactory(PlainTextKafkaAdminFactory plaintextAdminFactory, KafkaSecurityConfig securityConfig)
    {
        this.plaintextAdminFactory = requireNonNull(plaintextAdminFactory, "plaintextAdminFactory is null");
        requireNonNull(securityConfig, "securityConfig is null");
        securityProperties = new Properties();
        securityProperties.putAll(securityConfig.getKafkaClientProperties());
    }

    @Override
    public Properties configure(ConnectorSession session)
    {
        Properties properties = new Properties();
        properties.putAll(plaintextAdminFactory.configure(session));
        properties.putAll(securityProperties);
        return properties;
    }
}
```